src/mds/CInode.cc

```diff
@@ -3886,47 +3882,54 @@ void CInode::validate_disk_state(CInode::validated_data *results,
 
     bool _dirfrags(int rval) {
       // basic reporting setup
-      results->raw_rstats.checked = true;
-      results->raw_rstats.ondisk_read_retval = rval;
-      results->raw_rstats.passed = false; // we'll set it true if we make it
```

Comment:
You've dropped initializing this and it needs to start out as false or the validation will never report as failed!